Antlr4BuildTasks/Tasks/RunAntlrTool.cs

```diff
@@ -1261,7 +1261,7 @@ private void TestDataReceived(string data)
             }
             try
             {
-                Regex valid_version_pattern = new Regex("(OpenJDK Runtime Environment [^()]*[(]build (1[1-9]|[2-9][0-9]))|(Java[(]TM[)] SE Runtime Environment [(]build (1[1-9]|[2-9][0-9]))");
+                Regex valid_version_pattern = new Regex("(OpenJDK Runtime Environment (?:HomeBrew )?[^()]*[(]build (1[1-9]|[2-9][0-9]))|(Java[(]TM[)] SE Runtime Environment [(]build (1[1-9]|[2-9][0-9]))");
```

Comment:
Is this a magical string? Or can other labels also be used for altered versions of Java?

Reply:
Good question, @Corniel. No, it's not "magic", but it's what the ones that have created a Java package for Homebrew for macOS have put in there. I'm not familiar with many Java distros and packagings, so I don't know if there are many other variations here, I'm afraid.

I think this "checking if the java version is new enough" regex is a bit brittle too, but I'm not sure I have a brilliant idea for anything that is more robust. I'm not sure whether just removing the version check altogether would be better, or if that will open up a wormhole of other potential bugs

Reply:
I'm deferentially not an export on the subject (for starters, I'm a .NET developer), but when I looked at this I was just curious, hence my remark.

Reply:
I'm not sure why the maintainers of the [openjdk Homebrew formula](https://formulae.brew.sh/formula/openjdk) decided to change the version string. But, it did break this particular check here, though.

(I'm a .NET developer myself, btw 😆  )

@kaby76 - do you have any thoughts on whether it would be possible to merge this one?

Reply:
Will merge today. Sorry for the delay.

Reply:
Thanks a lot, @kaby76 - no need to say sorry, Open Source is mostly volunteer work, isn't it? No obligation to deliver quickly :) we're all thankful for the work that is done.